src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -938,5 +972,12 @@ public void Dispose()
 
             return _nodeSelector.GetCurrentNode();
         }
+
+        ~RequestExecutor()
+        {
+#if DEBUG
+            Debug.WriteLine("_nodeSelectorBySessionId is not empty when finalizer is called! Perhaps you somewhere missed calling session.Dipose()?");
```

Comment:
If you are doing that just for debug, make sure that in release builds, there isn't a finalizer at all.
Otherwise we pay for it in the finalizer queue.